tests/shardddl3/run.sh

```diff
@@ -873,6 +873,1442 @@ function DM_112() {
+         (partition p0 values less than (100), partition p1 values less than (200));\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_144_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10),(11),(12),(13),(14),(15),(110),(111),(112),(113),(114),(115);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(20),(21),(22),(23),(24),(25),(120),(121),(122),(123),(124),(125);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(30),(31),(32),(33),(34),(35),(130),(131),(132),(133),(134),(135);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} reorganize partition p0,p1 into (partition p0 values less than (200))"
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "query-status test" \
+        "ALTER TABLE \`${shardddl1}\`.\`${tb1}\` REORGANIZE PARTITION \`p0\`,\`p1\` INTO (PARTITION \`p0\` VALUES LESS THAN (200))" 1 \
+        "Unsupported reorganize partition" 1
+}
+
+# Reorganize partition.
+function DM_144 {
+    run_case 144 "double-source-pessimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id) \
+        (partition p0 values less than (100), partition p1 values less than (200));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id) \
+         (partition p0 values less than (100), partition p1 values less than (200));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (id int primary key) partition by range(id) \
+         (partition p0 values less than (100), partition p1 values less than (200));\"" \
+        "clean_table" "pessimistic"
+    run_case 144 "double-source-optimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id) \
+        (partition p0 values less than (100), partition p1 values less than (200));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id) \
+         (partition p0 values less than (100), partition p1 values less than (200));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (id int primary key) partition by range(id) \
+         (partition p0 values less than (100), partition p1 values less than (200));\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_145_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(100),(101),(102),(103),(104),(105);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(200),(201),(202),(203),(204),(205);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(300),(301),(302),(303),(304),(305);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} engine=innodb;"
+
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "query-status test" \
+        "This type of ALTER TABLE is currently unsupported" 1
```

Comment:
we might add a builtin rule to skip it 🤔

Reply:
I'm not sure what the builtin rule look like. Could you explain a little more?

Reply:
there're some RE rule https://github.com/pingcap/dm/blob/8cd3e8a4de975a76bb30528d7bf39a963a0ac2ee/pkg/utils/util.go#L41

and we could delete this spec when split multi-change DDL https://github.com/pingcap/dm/blob/8cd3e8a4de975a76bb30528d7bf39a963a0ac2ee/pkg/parser/common.go#L257

It's OK to change in another PR

Reply:
Oh I see, thx.